starks/poly_utils.py

```diff
@@ -63,8 +63,94 @@ def construct_affine_vanishing_polynomial(field: Field, aff: AffineSpace) -> Pol
   """Constructs a polynomial which vanishes over a given affine space."""
   # TODO(rbharath): Need to implement this correctly.
   aff_elts = [field(elt) for elt in aff]
+  ##############################################
+  # TODO(rbharath): This doesn't work for arbitrary finite fields! 
+  print("field")
+  print(field)
+  print("aff_elts")
+  print(aff_elts)
+  print("[type(aff) for aff in aff_elts]")
+  print([type(aff) for aff in aff_elts])
+  # Ok the error here is that aff_elts are not being interpreted as finite field elements? How to fix?
+  ##############################################
+  print(aff_elts)
   return zpoly(field, aff_elts)
 
+def construct_affine_vanishing_polynomial_Moore(field: Field, aff: AffineSpace) -> Poly:
+  """Constructs a polynomial which vanishes over a given affine space."""
+  b = aff.basis
+  Moore = []
+  for i in range(len(b)):
+    temp_l = []
+    temp_l.append(b[i])
+    for j in range(len(b)):
+      # I consider 2 for the pow because of binary finite field, 
+      temp_l.append(temp_l[0]**(2**(j+1)))
+    Moore.append(temp_l)
+
+  # compute Moore's nonzero kernel element
+  kernel = gauss(Moore, len(b), field)
+
+  # constructing the polynomial
+  polysOver = polynomials_over(field).factory
+  return polysOver(kernel)
+
+def gauss(M, row, field):
```

Comment:
Could you add test cases for the gaussian elimination on a couple simple matrices to check that it works correctly?